ui/ducks/send/send.js

```diff
@@ -2054,7 +2073,22 @@ export function editExistingTransaction(assetType, transactionId) {
     const transaction = unapprovedTransactions[transactionId];
     const account = getTargetAccount(state, transaction.txParams.from);
 
-    if (assetType === AssetType.native) {
+    const isSwapAndSend = state[name].prevSwapAndSendInput;
+
+    if (isSwapAndSend) {
+      dispatch(
+        actions.addNewDraft({
+          ...draftTransactionInitialState,
+          ...state[name].prevSwapAndSendInput,
+          id: transactionId,
+          fromAccount: account,
+          history: [
+            `sendFlow - user clicked edit on transaction with id ${transactionId} (swap and send)`,
+          ],
+        }),
+      );
+      dispatch(updateSendQuote());
+    } else if (assetType === AssetType.native) {
```

Comment:
Did you consider decoding the txData instead to extract the draft transaction details? The decoder can also be used to populate the recipient in the confirmation page (we currently show the swaps contract address)

Reply:
The swaps-recipient bug was a different that's since been resolved. However, you're 100% right about decoding being preferred. I realize that we lose the `prevSwapAndSendInput` whenever the extension is closed so that edge makes this worth implementing.

Reply:
Side note but it always amazes me how well you can pick up on what's going on in code – no doubt a huge contributor to the quality of your reviews. You're by far the best reviewer I've come across and I really do appreciate that ❤️